doc/source/conf.py

```diff
@@ -207,3 +207,11 @@
                    (None, 'http://data.astropy.org/intersphinx/matplotlib.inv')),
     'astropy': ('https://docs.astropy.org/en/stable/', None)
 }
+
+# Enable nitpicky mode - which ensures that all references in the docs resolve.
+nitpicky = True
+nitpick_ignore = [
+    ('py:class', 'stsci.tools.stpyfits._ConstantValueImageBaseHDU'),
+    ('py:class', 'astropy.io.fits.hdu.image.PrimaryHDU'),
+    ('py:class', 'astropy.io.fits.hdu.image.ImageHDU'),
```

Comment:
Instead of ignoring, use the proper API ref for astropy.io.fits objects.

https://docs.astropy.org/en/stable/io/fits/api/images.html#astropy.io.fits.ImageHDU

https://docs.astropy.org/en/stable/io/fits/api/hdus.html#astropy.io.fits.PrimaryHDU